frontend/public/components/graphs/area.tsx

```diff
@@ -35,13 +39,35 @@ export const AreaChart: React.FC<AreaChartProps> = ({
   query,
   theme = getCustomTheme(ChartThemeColor.blue, ChartThemeVariant.light, areaTheme),
   tickCount = DEFAULT_TICK_COUNT,
+  timespan,
   title,
   xAxis = true,
   yAxis = true,
 }) => {
   const [containerRef, width] = useRefWidth();
+  const [timestamp, setTimestamp] = React.useState(null);
+  React.useEffect(() => {
+    setTimestamp(Date.now());
+  }, [data])
+
   const getLabel = ({x, y}) => `${humanize(y).string} at ${formatDate(x)}`;
-  const container = <ChartVoronoiContainer voronoiDimension="x" labels={getLabel} />;
+  const container = <ChartVoronoiContainer labels={getLabel} voronoiDimension="x" />;
+  const max = data.length ? _.maxBy(data, 'y') : {};
+
+  // Force x-domain to show entire timespan
+  // Force y-domain to be a reasonable range for a single data point or if all data values are 0
+  const domain: DomainPropType = timestamp && {
+    ...timespan && { x: [
+      timestamp - timespan,
+      timestamp,
+    ]},
+    ...(data.length === 1 || max.y === 0) && {
```

Comment:
This won't do what we want if there are any negative `y` values.

Reply:
I just opened https://github.com/openshift/console/pull/2469, which makes some similar changes to the Query Browser Y-axis range.